clippy_lints/src/mem_replace.rs

```diff
@@ -43,6 +43,33 @@ declare_clippy_lint! {
     "replacing an `Option` with `None` instead of `take()`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `mem::replace()` on an `Option` with `Some(…)`.
+    ///
+    /// ### Why is this bad?
+    /// `Option` already has the method `replace()` for
+    /// taking its current value (Some(…) or None) and replacing it with
+    /// `Some(…)`.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// use std::mem;
```

Comment:
This is only used once, so we can inline the `std::` and remove the import to shorten the example.

Reply:
Done.